driven/storage/adapter.go

```diff
@@ -2543,11 +2536,72 @@ func (sa *Adapter) InsertAppOrgRole(context TransactionContext, item model.AppOr
 }
 
 //UpdateAppOrgRole updates application organization role
-func (sa *Adapter) UpdateAppOrgRole(item model.AppOrgRole) error {
-	//TODO
-	//This will be slow operation as we keep a copy of the entity in the users collection without index.
-	//Maybe we need to up the transaction timeout for this operation because of this.
-	return errors.New(logutils.Unimplemented)
+func (sa *Adapter) UpdateAppOrgRole(context TransactionContext, item model.AppOrgRole) error {
+	if context == nil {
```

Comment:
Why is context required to be passed in here?

Reply:
The context should be required because updating a role may require documents in multiple collections to be updated all at once. This operation should not be allowed unless done in a transaction.

Reply:
Shouldn't we establish a new context if none is passed rather than requiring that one is passed?

Reply:
Ok, yes, I added something in to handle it.